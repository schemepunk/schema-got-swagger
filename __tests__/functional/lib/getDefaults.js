'use strict';

const GetDefaults = require('../../../lib/GetDefaults');
const _ = require('lodash');
const { SchemaGotSwaggerError } = require('../../lib/SchemaGotSwaggerError');

let tmpMocks = [];

beforeEach(() => {
  tmpMocks.forEach(mock => mock.mockRestore());
  tmpMocks = [];
  jest.resetAllMocks();
  jest.spyOn(Date, 'now').mockReturnValue(2000);
});

afterAll(() => {
  jest.restoreAllMocks();
});

describe('Get Defaults', () => {
  test('Get existing yaml file', () => {
    expect.assertions(1);
    const getDefaults = new GetDefaults('Sgs');
    return getDefaults.getDefaults()
    .then(data => expect(data).toMatchSnapshot());
  });
  test('Get function based defaults', () => {
    expect.assertions(1);
    const getDefaults = new GetDefaults('SwaggerMainTemplates', '2.0.0');
    return getDefaults.getDefaults()
    .then(data => expect(data).toMatchSnapshot());
  });
  test('Get function based defaults', () => {
    expect.assertions(1);
    expect(() => new GetDefaults('nope')).toThrow();
  });
});
