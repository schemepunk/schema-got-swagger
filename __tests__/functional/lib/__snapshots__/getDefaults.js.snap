// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Get Defaults Get existing yaml file 1`] = `
Object {
  "apiType": "jsonApi",
  "includePathsInDefinitions": false,
  "mainSwaggerSchemeProcessName": "processTemplates",
  "pathsSchemeProcessName": "processTemplates",
  "realizationsSource": "swaggerSrc",
  "sgsSemver": Object {
    "composer": Object {
      "composerType": "default",
      "destination": "",
      "priority": "default",
    },
    "converterType": "default",
    "directoryFileIgnorePattern": "json$",
    "groups": Object {
      "example": Object {
        "members": Array [
          "test",
        ],
      },
    },
    "prereleaseOrdering": Object {},
    "semverishObjectLocation": "",
    "semveristBehaviors": Object {
      "default": false,
      "defaultName": "default",
      "groups": false,
      "inheritence": "semverImplied",
      "lazySemverist": Object {
        "attributes": false,
        "preReleaseForwards": false,
      },
      "mergeStrategy": "merge",
      "preReleasePattern": "[a-z]",
    },
  },
  "sgsType": "semver",
  "swaggerVersion": "3.0.0",
}
`;

exports[`Get Defaults Get function based defaults 1`] = `
Object {
  "basePath": "{{apiBasePath}}",
  "consumes": "[{{#consumes}}\\"{{mime.type}}\\"{{#mime.comma}}, {{/mime.comma}}{{/consumes}}]",
  "contact": "{{=<% %>=}}{<%={{ }}=%> \\"name\\": \\"{{contact.name}}\\", \\"url\\": \\"{{&contact.url}}\\", \\"email\\": \\"{{contact.email}}\\" {{=<% %>=}}}<%={{ }}=%>",
  "definitions": "{{#definitions}}{{&process}}{{/definitions}}",
  "description": "{{apiDescription}}",
  "destinationTemplate": "{{=<% %>=}}{<%={{ }}=%>\\"swagger\\": \\"2.0\\", \\"info\\": {{=<% %>=}}{<%={{ }}=%> \\"title\\": \\"{{> title}}\\", {{#description}}\\"description\\": \\"{{> description}}\\",{{/description}} {{#termsOfService}}\\"termsOfService\\": \\"{{> termsOfService}}\\",{{/termsOfService}} \\"version\\": \\"{{> version}}\\", {{#contact}}\\"contact\\": {{> contact}},{{/contact}} {{#license}}\\"license\\": {{> license}}{{/license}}{{=<% %>=}}}<%={{ }}=%>, {{#host}}\\"host\\": \\"{{> host}}\\",{{/host}} {{#basePath}}\\"basePath\\": \\"{{> basePath}}\\",{{/basePath}} {{#schemes}}\\"schemes\\": {{> schemes}}, {{/schemes}} {{#consumes}}\\"consumes\\": {{> consumes}}, {{/consumes}} {{#produces}}\\"produces\\": {{> produces}}, {{/produces}} \\"paths\\": {}, {{#definitions}}\\"definitions\\": {{> definitions}},{{/definitions}} {{#parameters}}\\"parameters\\": {{> parameters }},{{/parameters}} {{#responses}}\\"responses\\": {{> responses }},{{/responses}} {{#securityDefinitions}}\\"securityDefinitions\\": {{> securityDefinitions}},{{/securityDefinitions}} {{#security}}\\"security\\": {{> security}},{{/security}} {{#tags}}\\"tags\\": {{> tags}},{{/tags}} {{#externalDocs}}\\"externalDocs\\": {{> externalDocs}}{{/externalDocs}} {{=<% %>=}}}<%={{ }}=%>",
  "externalDocs": "{{#externalDocs}}{{=<% %>=}}{<%={{ }}=%>{{#description}}\\"description\\": \\"{{description}}\\", {{/description}}\\"url\\": \\"{{&url}}\\"{{=<% %>=}}}<%={{ }}=%>{{/externalDocs}}",
  "host": "{{apiHost}}",
  "license": "{{=<% %>=}}{ \\"name\\": \\"<%license.name%>\\", \\"url\\": \\"<%&license.url%>\\"}<%={{ }}=%>",
  "parameters": "{{#parameters}}{{&process}}{{/parameters}}",
  "produces": "[{{#produces}}\\"{{mime.type}}\\"{{#mime.comma}}, {{/mime.comma}}{{/produces}}]",
  "responses": "{{#responses}}{{&process}}{{/responses}}",
  "schemes": "{{#schemes}}{{&process}}{{/schemes}}",
  "security": "{{#security}}{{&process}}{{/security}}",
  "securityDefinitions": "{{#securityDefinitions}}{{&process}}{{/securityDefinitions}}",
  "tags": "{{#tags}}{{&process}}{{/tags}}",
  "termsOfService": "{{&termsOfService}}",
  "title": "{{apiName}}",
  "version": "{{#apiVersion}}{{apiVersion}}{{/apiVersion}}{{^apiVersion}}{{apiSemver}}{{/apiVersion}}",
}
`;
